### PR TITLE
simplify IEndpointRouter interface

### PR DESCRIPTION
**Pull request by vicancy**

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Common/Endpoints/DefaultEndpointRouter.cs [hunk 1]
```diff
@@ -9,82 +9,82 @@ namespace Microsoft.Azure.SignalR
...
-        /// Broadcast to all available endpoints
+        /// Only primary endpoints will be returned by client /negotiate
+        /// If no primary endpoint is available, promote one secondary endpoint
         /// </summary>
-        /// <param name="connectionId"></param>
-        /// <param name="availableEndpoints"></param>
-        /// <returns></returns>
-        public IEnumerable<ServiceEndpoint> GetEndpointsForConnection(string connectionId, IEnumerable<ServiceEndpoint> availableEndpoints)
+        /// <returns>The availbale endpoints</returns>
+        private ServiceEndpoint[] GetPrimaryEndpoints(IEnumerable<ServiceEndpoint> endpoints)
```
Comment (chenkennt) [inline src/Microsoft.Azure.SignalR.Common/Endpoints/DefaultEndpointRouter.cs]: protected?

Comment (chenkennt) [reply src/Microsoft.Azure.SignalR.Common/Endpoints/DefaultEndpointRouter.cs]: PrimaryEndpoints may be misleading here.

File: src/Microsoft.Azure.SignalR.Common/Interfaces/IEndpointRouter.cs [hunk 2]
```diff
@@ -8,22 +8,18 @@ namespace Microsoft.Azure.SignalR
...
+        /// Get the service endpoint for the client to connect to
         /// </summary>
-        /// <param name="primaryEndpoints"></param>
+        /// <param name="endpoints"></param>
         /// <returns></returns>
-        ServiceEndpoint GetNegotiateEndpoint(IEnumerable<ServiceEndpoint> primaryEndpoints);
+        ServiceEndpoint GetNegotiateEndpoint(IEnumerable<ServiceEndpoint> endpoints);
 
-        IEnumerable<ServiceEndpoint> GetEndpointsForBroadcast(IEnumerable<ServiceEndpoint> availableEndpoints);
+        IEnumerable<ServiceEndpoint> GetEndpointsForBroadcast(IEnumerable<ServiceEndpoint> endpoints);
```
Comment (chenkennt) [inline src/Microsoft.Azure.SignalR.Common/Interfaces/IEndpointRouter.cs]: Also add some comments for these methods

File: src/Microsoft.Azure.SignalR.Common/Endpoints/DefaultEndpointRouter.cs [hunk 3]
```diff
@@ -3,88 +3,90 @@
 
 using System.Collections.Generic;
 using System.Linq;
+using Microsoft.Azure.SignalR.Common;
 
 namespace Microsoft.Azure.SignalR
 {
     public class DefaultEndpointRouter : IEndpointRouter
```
Comment (vwxyzh) [inline src/Microsoft.Azure.SignalR.Common/Endpoints/DefaultEndpointRouter.cs]: Decorator?

**Pull request merged**